src/inmanta/deploy/persistence.py

```diff
@@ -180,14 +185,10 @@ async def send_in_progress(
 
                 # FIXME: we may want to have this in the RPS table instead of Resource table, at some point
                 await resource.update_fields(connection=connection, status=const.ResourceState.deploying)
-
-            # FIXME: shortcut to use scheduler state
-            result = await data.Resource.get_last_non_deploying_state_for_dependencies(
-                environment=self.environment, resource_version_id=resource_id_parsed, connection=connection
+            return await self.scheduler.get_last_non_deploying_state_for_dependencies(
+                resource_id=resource_id_parsed.resource_str()
```

Comment:
I don't think this belongs here. It makes the flow a bit more convoluted than it needs to be: `scheduler -> task -> scheduler -> update manager -> scheduler` while it could simply be `scheduler -> task -> scheduler`. Most importantly, it adds a lock on yet another layer. I think we could easily make this part of `Scheduler.get_resource_intent` instead. It doesn't really have anything to do with persistence anymore.

Reply:
This will also solve the cyclic import issue.